proxy/src/control/discovery.rs

```diff
@@ -55,8 +68,26 @@ pub struct DiscoveryWork<T: HttpService<ResponseBody = RecvBody>> {
     rx: mpsc::UnboundedReceiver<(FullyQualifiedAuthority, mpsc::UnboundedSender<Update>)>,
 }
 
+#[derive(Clone, Debug, Hash, Eq, PartialEq)]
+pub struct DstLabels(Arc<str>);
+
+/// Any additional metadata describing a discovered service.
+#[derive(Clone, Debug, Hash, Eq, PartialEq)]
+pub struct Metadata {
+    /// A set of Prometheus metric labels describing the destination.
+    metric_labels: Option<DstLabels>,
+}
+
+/// Middleware that adds an extension containing an optional set of metric
+/// labels to requests.
+#[derive(Clone, Debug)]
+pub struct Labeled<T> {
+    metric_labels: Option<futures_watch::Watch<Option<DstLabels>>>,
+    inner: T,
+}
```

Comment:
All of these types seem like they have more to do with telemetry than they have to do with service discovery. Accordingly, I think it would be better to put them in the telemetry and/or label modules instead of here.

Reply:
Moved in aae6527fca40580a2678fa794b53931c329638b7 --- good point!